From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 17 Sep 2018 01:11:22 +0100
Subject: mm: Avoid ABI change in 4.18.7
Forwarded: not-needed

Commit 8bfd9029bc79 "powerpc/64s: Fix page table fragment refcount race
vs speculative references" introduced a union with another alternate use
for one of the words in struct page.

The layout of the structure is unchanged, and this use is private to
the powerpc page table allocator, so it's not actually an ABI change.
Therefore hide it from genksyms.
---
--- a/include/linux/mm_types.h
+++ b/include/linux/mm_types.h
@@ -139,10 +139,14 @@ struct page {
 			unsigned long _pt_pad_1;	/* compound_head */
 			pgtable_t pmd_huge_pte; /* protected by page->ptl */
 			unsigned long _pt_pad_2;	/* mapping */
+#ifndef __GENKSYMS__
 			union {
 				struct mm_struct *pt_mm; /* x86 pgds only */
 				atomic_t pt_frag_refcount; /* powerpc */
 			};
+#else
+			struct mm_struct *pt_mm; /* x86 pgds only */
+#endif
 #if ALLOC_SPLIT_PTLOCKS
 			spinlock_t *ptl;
 #else
